ntcore/src/main/native/include/networktables/StructTopic.h

```
@@ -0,0 +1,387 @@
+   */
+  ValueType Get() const { return Get(m_defaultValue); }
+
+  /**
+   * Get the last published value.
+   * If no value has been published, returns the passed defaultValue.
+   *
+   * @param defaultValue default value to return if no value has been published
+   * @return value
+   */
+  ValueType Get(const T& defaultValue) const {
+    return GetAtomic(defaultValue).value;
+  }
+
+  /**
+   * Get the last published value along with its timestamp
+   * If no value has been published, returns the stored default value and a
+   * timestamp of 0.
+   *
+   * @return timestamped value
+   */
+  TimestampedValueType GetAtomic() const { return GetAtomic(m_defaultValue); }
+
+  /**
+   * Get the last published value along with its timestamp.
+   * If no value has been published, returns the passed defaultValue and a
+   * timestamp of 0.
+   *
+   * @param defaultValue default value to return if no value has been published
+   * @return timestamped value
+   */
+  TimestampedValueType GetAtomic(const T& defaultValue) const {
+    wpi::SmallVector<uint8_t, S::kSize> buf;
+    TimestampedRawView view = ::nt::GetAtomicRaw(m_subHandle, buf, {});
+    if (view.value.size() < S::kSize) {
+      return {0, 0, defaultValue};
+    } else {
+      return {view.time, view.serverTime,
+              S::Unpack(view.value.subspan<0, S::kSize>())};
+    }
+  }
+
+  /**
+   * Get an array of all value changes since the last call to ReadQueue.
+   * Also provides a timestamp for each value.
+   *
+   * @note The "poll storage" subscribe option can be used to set the queue
+   *     depth.
+   *
+   * @return Array of timestamped values; empty array if no new changes have
+   *     been published since the previous call.
+   */
+  std::vector<TimestampedValueType> ReadQueue() {
+    auto raw = ::nt::ReadQueueRaw(m_subHandle);
+    std::vector<TimestampedValueType> rv;
+    rv.reserve(raw.size());
+    for (auto&& r : raw) {
+      if (r.value.size() < S::kSize) {
+        continue;
```

Is this desired behavior (completely and silently skipping over too-small values)? Should this be documented?

---

I think documentation + single print would be fine (unless we're worried about the flag's memory).